src/App.tsx

```diff
@@ -2,24 +1,53 @@
+import React, { ChangeEvent, useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const items : string[] = getNumbers(1, 42).map(n => `Item ${n}`);
+
+enum PerPageOption {
+  'three' = 3,
+  'five' = 5,
+  'ten' = 10,
+  'twenty' = 20
+}
 
 export const App: React.FC = () => {
+  const [perPageSelector, setPerPageSelector] = useState<PerPageOption>(PerPageOption.five);
+  const [activePage, setActivePage] = useState<number>(1);
+
+  function handleSelect (e: ChangeEvent<HTMLSelectElement>) {
+    setPerPageSelector(+e.currentTarget.value)
+    setActivePage(1)
+  }
+
+  function changePage (page: number) {
+    setActivePage(page);
+  }
+
+  const paginationButtonsLength = Math.ceil(items.length / perPageSelector);
+
+  const to = activePage * perPageSelector;
+  const from = to - (perPageSelector - 1);
```

Comment:
The calculation for `from` should be `const from = to - perPageSelector + 1;` to correctly calculate the starting index of the items to display. This ensures the correct range of items is shown on each page.